### ISSUE TITLE
Mark RobustDeterminant as deprecated

### ISSUE THREAD

**Issue by FObermaier** [locationtech/jts#267]
I understand the commit 886a389e525aca1022ba765a146c85b40ebf8f36 as if it is intended to remove `org.locationtech.jts.algorithm.RobustDeterminant` at some point. If so, it should be marked `@deprecated`

**Comment by jnh5y** [comment]
Good catch.  If we do deprecate and remove this, we should probably chat with @jodygarnett about how GeoTools will be impacted...

**Comment by dr-jts** [comment]
For reference, this is referring to PR #216

**Comment by dr-jts** [comment]
Agreed, should check and see if marking as deprecated will cause issues with downstream usage.  Marking things as @deprecated can cause errors or warnings in projects that are still using them.  

In this case a "soft" deprecation could be indicated, via Javadoc comments.
